### PR TITLE
EVG-5226 prevent duplicate task definition in a variant

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

**Comment by john-m-liu** [comment]
If someone is currently trying this, it wouldn't be working because all tasks would receive the same ID, and so only 1 would ever exist in each version. If we want to support that setup we will need to make additional changes

File: model/task/sorter.go [hunk 1]
```diff
@@ -1,12 +1,8 @@
...
 
-import (
-	"github.com/evergreen-ci/evergreen/db"
-	"github.com/mongodb/grip"
-	"github.com/mongodb/grip/message"
-)
+import "github.com/evergreen-ci/evergreen/db"
 
-type Tasks []Task
+type Tasks []*Task
```
Comment (tychoish) [inline model/task/sorter.go]: why do we need this? are we setting ourselves up for a range-over-slice-of-pointers bug?

Comment (tychoish) [reply model/task/sorter.go]: I would unwind if possible.

**Comment by john-m-liu** [comment]
Ran this locally and did not find any projects that are violating this currently (server fix this over the weekend)

File: model/task/sorter.go (hunk 1 above)
Comment (john-m-liu) [reply model/task/sorter.go]: This was the definition before - my changes in this file are a revert

**Pull request merged**